app/database/result_store_propagate.go

```diff
@@ -302,9 +301,8 @@ func (s *ResultStore) Propagate() (err error) {
 				can_enter_from = IF(
 					VALUES(can_enter_from) <> '9999-12-31 23:59:59' AND can_enter_from > VALUES(can_enter_from),
 					VALUES(can_enter_from), can_enter_from),
-				can_enter_until = IF(
-					VALUES(can_enter_from) <> '9999-12-31 23:59:59' AND can_enter_until <> '9999-12-31 23:59:59',
-					'9999-12-31 23:59:59', can_enter_until)`, canViewContentIndex, canViewContentIndex)
+				can_enter_until = IF(VALUES(can_enter_from) <> '9999-12-31 23:59:59', '9999-12-31 23:59:59', can_enter_until)`,
```

Comment:
But so, if the new value of `from`is +inf, you set `until` to its existing value? It looks like the case where you want to put `until` to `+inf` anyway. (`until` < `from` makes no sense).

The old rule meant: we set `until` to `+inf` EXCEPT if it previous value was not `+inf`and the new `from` is `+inf`... in which case we leave it to its previous value. (so `until` < `from` )

But as this `until` < `from`  does not make sense and "item_unlocking" rules are only set by code that we control, I think, we can just ignore `can_enter_until` and leave it to its db default value "9999-12-31 23:59:59".

Reply:
In this query we never want to set `can_enter_until` to +inf explicitly.

Actually, the old rule meant we set until to +inf EXCEPT if it previous value was not +inf OR the new from is +inf... in which case we leave it to its previous value.

The only difference between the old rule and the new rule is that the old rule explicitly kept the old value of `can_enter_until` if it was +inf. The new rule overwrites +inf by +inf. So there is no difference between the rules.

Reply:
I just mean, that the new like the new rule does not really sense IMO. So we may not set `can_enter_until` at all.

Reply:
But if the can_enter_until is in the past, the item will not be unlocked (no matter how we set can_enter_from), so we should update can_enter_until here.

Reply:
Why would it be in the past for a "item_unlocking" row? But so, put it to "9999-12-31 23:59:59" without conditions.

Reply:
fixed